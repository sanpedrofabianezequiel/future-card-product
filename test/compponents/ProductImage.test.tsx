import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('Product Image', () => {

    test('debe de mostrar el componente correctamente con la imagen personalizado', () => {

        const wrapper = renderer.create(
            <ProductImage  img='https://hola.jpg' />
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
    })
    
    test('debe de mostrar el componente con la imagen del producto', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2} >
                {
                    ()=>(
                        <ProductImage />
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    })
    
})
