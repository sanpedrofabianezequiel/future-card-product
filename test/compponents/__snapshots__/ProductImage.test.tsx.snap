// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Image debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Product Image debe de mostrar el componente correctamente con la imagen personalizado 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;
